lib/routes/veeam/get.js

```diff
@@ -25,21 +26,44 @@ function getVeeamFile(request, response, bucketMd, log) {
         if (err) {
             return responseXMLBody(errors.InternalError, null, response, log);
         }
-        const fileToBuild = getFileToBuild(request, data._capabilities?.VeeamSOSApi);
-        if (fileToBuild.error) {
-            return responseXMLBody(fileToBuild.error, null, response, log);
-        }
-        let modified = new Date().toISOString();
-        // Extract the last modified date, but do not include it when computing
-        // the file's ETag (md5)
-        modified = fileToBuild.value.LastModified;
-        delete fileToBuild.value.LastModified;
+        const finalizeRequest = bucketMetrics => {
+            const fileToBuild = getFileToBuild(request, data._capabilities?.VeeamSOSApi);
+            if (fileToBuild.error) {
+                return responseXMLBody(fileToBuild.error, null, response, log);
+            }
+
+            // Extract the last modified date, but do not include it when computing
+            // the file's ETag (md5)
+            const modified = fileToBuild.value.LastModified;
+            delete fileToBuild.value.LastModified;
 
-        const builder = new xml2js.Builder({
-            headless: true,
-        });
-        return respondWithData(request, response, log, data,
-            buildHeadXML(builder.buildObject(fileToBuild.value)), modified);
+            // The SOSAPI metrics are dynamically computed using the SUR backend.
+            if (bucketMetrics && !fileToBuild.value.CapacityInfo?.Used) {
+                fileToBuild.value.CapacityInfo.Used = Number(bucketMetrics.bytesTotal);
+                fileToBuild.value.CapacityInfo.Available =
+                    Number(fileToBuild.value.CapacityInfo.Capacity) - Number(bucketMetrics.bytesTotal);
+                // TODO CLDSRV-633 when SUR backend supports realtime metrics: enable this, it will
+                // report the real last modified date of the bucket, instead of the current date,
+                // ensuring no issue in a SOSAPI context.
+                // modified = new Date(bucketMetrics.date).toISOString();
+            }
+
+            const builder = new xml2js.Builder({
+                headless: true,
+            });
+            return respondWithData(request, response, log, data,
+                buildHeadXML(builder.buildObject(fileToBuild.value)), modified);
+        };
+        if (!isSystemXML(request.objectKey)) {
```

Comment:
Should we also check that the scuba backend is enabled ?

Reply:
Practically this won't change much: in this branch and the branch before, scuba is always deployed. Inflight quota service is enabled based on the scuba presence in the config. here, having no scuba would just cause an error we already handle. We do not want to default to zero metrics when there is no Scuba, so we can only return a more specific error.